File: task_1/prime_numbers.js
const userNumber = process.argv[2];
console.time();
let answer = primeNumbers (userNumber);
console.timeEnd();
console.log(answer);


function primeNumbers (amountNumbers) {
    if (isNaN(amountNumbers) || amountNumbers <= 0) {
        return "Вы ошиблись при вводе параметра!";
    } 
    let result = [2,];
    let number = 2;
    while (result.length < amountNumbers){
        number++;
        let counter = 0;
        for (let denominator of result) {
            if (number % denominator === 0) {
                counter++;
                break;
            }
        }
        if (counter === 0) {
            result[result.length] = number;
        }
    }
    return result;
}

File: task_1/prime_numbers.py
from pprint import pprint
import time


def prime_numbers(amount_numbers: int | str) -> list | str:
    """
    Вычисляет и выводит в виде списка указанное в параметре количество простых чисел

    :param amount_numbers: Требуемое количество простых чисел в виде числа или строки
    :return: Список простых чисел
    """
    if not amount_numbers.isdigit() or int(amount_numbers) <= 0:
        return "Вы ошиблись при вводе параметра!\n"
    if isinstance(amount_numbers, str):
        amount_numbers = int(amount_numbers)
    result = [2, ]
    number = 2
    while len(result) < amount_numbers:
        number += 1
        counter = 0
        for denominator in result:
            if number % denominator == 0:
                counter += 1
                break
        if counter == 0:
            result.append(number)
    return result


if __name__ == "__main__":
    user_number = (input("Введите необходимое количество простых чисел!\n")).strip()
    start = time.perf_counter()
    answer = prime_numbers(user_number)
    end = time.perf_counter()
    pprint(answer, compact=True)
    print(f'Вычисления заняли {end - start}')
